tests/jballerina-unit-test/src/test/resources/test-src/javainterop/dependently_typed_functions_test_negative.bal

```diff
@@ -370,3 +370,15 @@ function testDependentlyTypedFunctionWithInferredArgForParamOfTypeReferenceTypeN
 
     var _ = functionWithInferredArgForParamOfTypeReferenceType(boolean);
 }
+
+function getArrayInferred(typedesc<anydata> td = <>) returns td[] = @java:Method {
+    'class: "org.ballerinalang.nativeimpl.jvm.tests.VariableReturnType",
+    name: "getArray",
+    paramTypes: ["io.ballerina.runtime.api.values.BTypedesc"]
+} external;
+
+type AnyArray any[];
+
+function testDependentlyTypedFunctionWithTypeReferenceTypeNegative() returns error? {
+    AnyArray _ = check getArrayInferred();
```

Comment:
Shall we add a test with union types as the expected type for confirm the behaviour.

ex:- `type C A|B`

and C as the expected type

Reply:
This issue arises when a member of the union is a type reference type, that is covered with the usage of check when the expected type is a type reference type - effectively `IntArray|error`.